File: CourierAmexAPI/CourierAmex/Models/ZoneModel.cs
namespace CourierAmex.Models
{
    public class ZoneModel : BaseEntityModel<int>
    {
        public required int CountryId { get; set; }
        public required int StateId { get; set; }
        public required string Code { get; set; }
        public required string Name { get; set; }
        public string? Notes { get; set; }
        public int Route { get; set; }

        public string? CountryName { get; set; }
        public string? StateName { get; set; }
    }
}

File: CourierAmexClient/src/app/features/general/models/zone.model.ts
import { BaseEntity } from "@app/models";

export interface ZoneModel extends BaseEntity<number> {
  countryId?: number;
  stateId?: number;
  code: string;
  name: string;
  notes?: string;
  route?: number;

  countryName?: string;
  stateName?: string;
}

export const newZoneModel: ZoneModel = {
  id: 0,
  countryId: undefined,
  stateId: undefined,
  code: '',
  name: '',
  status: 2
}
